packages/auth/amplify_auth_cognito_dart/test/plugin/auth_providers_test.dart

```diff
@@ -29,15 +30,46 @@ AWSHttpRequest _generateTestRequest() {
   );
 }
 
-/// Returns dummy AWS credentials.
+/// Returns dummy AWS credentials and access token.
 class TestAmplifyAuth extends AmplifyAuthCognitoDart {
   @override
   Future<AuthSession> fetchAuthSession({
     required AuthSessionRequest request,
   }) async {
-    return const CognitoAuthSession(
+    return CognitoAuthSession(
       isSignedIn: true,
-      credentials: AWSCredentials('fakeKeyId', 'fakeSecret'),
+      credentials: const AWSCredentials('fakeKeyId', 'fakeSecret'),
+      userPoolTokens: CognitoUserPoolTokens(
+        accessToken: accessToken,
+        idToken: idToken,
+        refreshToken: refreshToken,
+      ),
+    );
+  }
+}
+
+/// Mock implementation of user pool only error when trying to get credentials.
+class TestAmplifyAuthUserPoolOnly extends AmplifyAuthCognitoDart {
```

Comment:
Can you follow [this](https://github.com/aws-amplify/amplify-flutter/blob/next/packages/auth/amplify_auth_cognito_dart/test/plugin/sign_out_test.dart#L218) pattern? If you override `fetchAuthSession` you're only testing the auth provider. If you use the default impl, you're testing the whole stack and this becomes an integration test which is much more valuable for us, since we will know if any change to fetchAuthSession breaks this model.

Reply:
Sure, revised this to change mock implementation like this.